modin/api/partition_api.py

```diff
@@ -78,3 +81,83 @@ def _unwrap_partitions(oid):
             part.coalesce(bind_ip=bind_ip).unwrap(squeeze=True, bind_ip=bind_ip)
             for part in partitions
         ]
+
+
+def create_df_from_partitions(partitions, axis):
+    """
+    Create DataFrame from remote partitions.
+
+    Parameters
+    ----------
+    partitions : list
+        List of Ray.ObjectRef/Dask.Future referencing to partitions in depend of the engine used.
+        Or list containing tuples of Ray.ObjectRef/Dask.Future referencing to ip addresses of partitions
+        and partitions itself in depend of the engine used.
+    axis : None, 0 or 1
+        The `axis` parameter is used to identify what are the partitions passed.
+        You have to set:
+        - `axis` to 0 if you want to create DataFrame from row partitions.
+        - `axis` to 1 if you want to create DataFrame from column partitions.
+        - `axis` to None if you want to create DataFrame from 2D list of partitions.
+
+    Returns
+    -------
+    DataFrame
+        DataFrame instance created from remote partitions.
+    """
+    from modin.data_management.factories.dispatcher import EngineDispatcher
+
+    factory = EngineDispatcher.get_engine()
+
+    partition_class = factory.io_cls.frame_cls._frame_mgr_cls._partition_class
+    partition_frame_class = factory.io_cls.frame_cls
+    partition_mgr_class = factory.io_cls.frame_cls._frame_mgr_cls
+
+    if axis is None:
+        if isinstance(partitions[0][0], tuple):
+            parts = np.array(
+                [
+                    [partition_class(partition, ip=ip) for ip, partition in row]
+                    for row in partitions
+                ]
+            )
+        else:
+            parts = np.array(
+                [
+                    [partition_class(partition) for partition in row]
+                    for row in partitions
+                ]
+            )
+    else:
+        if axis == 0:
+            if isinstance(partitions[0], tuple):
+                parts = np.array(
+                    [
+                        [partition_class(partition, ip=ip)]
+                        for ip, partition in partitions
+                    ]
+                )
+            else:
+                parts = np.array(
+                    [[partition_class(partition)] for partition in partitions]
+                )
+        elif axis == 1:
+            if isinstance(partitions[0], tuple):
+                parts = np.array(
+                    [
+                        [
+                            partition_class(partition, ip=ip)
+                            for ip, partition in partitions
+                        ]
+                    ]
+                )
+            else:
+                parts = np.array(
+                    [[partition_class(partition) for partition in partitions]]
+                )
```

Comment:
there should be a global `else` clause when `axis` is neither `None` or `0` or `1`, with a `ValueError` probably

Reply:
I don't think that somebody will pass different value of axis to the function. But fixed anyway.